src/covid_model_seiir_pipeline/forecasting/task/beta_forecast.py

```diff
@@ -26,99 +26,118 @@ def run_beta_forecast(draw_id: int, forecast_version: str, scenario_name: str):
-    forecasts = []
-    for location_id in location_ids:
-        log.info(f"On location id {location_id}")
-        init_cond = initial_condition.loc[location_id].values
-        total_population = init_cond.sum()
-
-        model_specs = model.SeiirModelSpecs(
-            alpha=beta_params['alpha'],
-            sigma=beta_params['sigma'],
-            gamma1=beta_params['gamma1'],
-            gamma2=beta_params['gamma2'],
-            N=total_population,
-        )
-        ode_runner = model.ODERunner(model_specs, init_cond)
-
-        loc_betas = betas.loc[location_id].sort_values('date')
-        loc_days = loc_betas['date']
-        loc_times = np.array((loc_days - loc_days.min()).dt.days)
-        loc_betas = loc_betas['beta_pred'].values
-        loc_thetas = np.repeat(thetas.get(location_id, default=0), loc_betas.size)
-
-        forecasted_components = ode_runner.get_solution(loc_times, loc_betas, loc_thetas)
-        forecasted_components['date'] = loc_days.values
-        forecasted_components['location_id'] = location_id
-        forecasts.append(forecasted_components)
-
-    forecasts = pd.concat(forecasts)
-    # Concat past with future
-    shared_columns = ['date', 'S', 'E', 'I1', 'I2', 'R', 'beta']
-    components = (pd.concat([beta_regression_df.loc[~transition_day, shared_columns].reset_index(),
-                             forecasts[['location_id'] + shared_columns]])
-                    .sort_values(['location_id', 'date'])
-                    .set_index(['location_id']))
-    components['theta'] = thetas.reindex(components.index).fillna(0)
-    data_interface.save_components(components.reset_index(), scenario_name, draw_id)
+    beta_scales = data_interface.load_beta_scales(scenario=scenario_name, draw_id=draw_id)
+
+    # We'll use the same params in the ODE forecast as we did in the fit.
+    beta_params = data_interface.load_beta_params(draw_id=draw_id)
 
+    # We'll need this to compute deaths and to splice with the forecasts.
     infection_data = data_interface.load_infection_data(draw_id)
 
-    observed_infections, modeled_infections = model.compute_infections(infection_data, components)
-    infections = observed_infections.combine_first(modeled_infections)['cases_draw'].rename('infections')
+    # Modeling starts
+    betas = model.forecast_beta(covariate_pred, coefficients, beta_scales)
+    future_components = model.run_normal_ode_model_by_location(initial_condition, beta_params, betas, thetas,
+                                                               location_ids, scenario_spec.solver)
+    components = model.splice_components(past_components, future_components, transition_day)
+    components['theta'] = thetas.reindex(components.index).fillna(0)
+    infections, deaths, r_effective = model.compute_output_metrics(infection_data, components, beta_params)
+
+    if scenario_spec.algorithm == 'mandate_reimposition':
+        min_wait = pd.Timedelta(days=7)
+        days_on = pd.Timedelta(days=7 * 6)
```

Comment:
Should this be `min_wait * 6` or is it just coincidence they're both in units of 1 week?

Reply:
coincidence